From e1d9125dd37ba05a1beaf5d565393c9239328c11 Mon Sep 17 00:00:00 2001
From: Martin Alejandro Oviedo <quamagi@hotmail.com>
Date: Thu, 30 Jul 2026 15:04:55 -0300
Subject: [PATCH] Integrate Machine Care policy into kernel Kconfig and
 Makefile

Add CONFIG_SPASM_KERNEL_CARE_POLICY to lib/Kconfig with dependency
on SPASM_KERNEL_GLOBAL_MIGRATION and default y for SpASM/Dual modes.
Add spasm_care_policy.o to lib/Makefile build rules.
---
 lib/Kconfig  | 11 +++++++++++
 lib/Makefile |  1 +
 2 files changed, 12 insertions(+)

diff --git a/lib/Kconfig b/lib/Kconfig
index cb7d353b1..36d9cb948 100644
--- a/lib/Kconfig
+++ b/lib/Kconfig
@@ -72,6 +72,17 @@ config SPASM_KERNEL_HEX_TO_BIN_DUAL
 
 endchoice
 
+config SPASM_KERNEL_CARE_POLICY
+	bool "spasm-kernel Machine Care policy"
+	depends on SPASM_KERNEL_GLOBAL_MIGRATION
+	default y if SPASM_KERNEL_MODE_SPASM
+	default y if SPASM_KERNEL_MODE_DUAL
+	help
+	  Compile the Machine Care policy engine written in SpASM.
+	  This provides spasm_care_level_v1() which maps CPU usage,
+	  temperature, and free RAM to a system care level
+	  (GREEN, YELLOW, ORANGE, RED, EMERGENCY).
+
 config RAID6_PQ
 	tristate
 
diff --git a/lib/Makefile b/lib/Makefile
index 3abd62743..c4d763ec6 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -65,6 +65,7 @@ obj-$(CONFIG_SPASM_KERNEL_BCD_SPASM) += bcd.o bcd2bin_spasm.o bcd2bin_spasm_entr
 obj-$(CONFIG_SPASM_KERNEL_BCD_DUAL) += bcd.o bcd2bin_spasm.o bcd2bin_dual.o bin2bcd_spasm.o bin2bcd_dual.o
 
 obj-$(CONFIG_SPASM_KERNEL_HEX_TO_BIN_SPASM) += hex_to_bin_spasm.o hex_to_bin_spasm_entry.o hex_to_bin_export.o
+obj-$(CONFIG_SPASM_KERNEL_CARE_POLICY) += spasm_care_policy.o
 obj-$(CONFIG_TEST_HEXDUMP) += test_hexdump.o
 obj-y += kstrtox.o
 obj-$(CONFIG_FIND_BIT_BENCHMARK) += find_bit_benchmark.o
-- 
2.47.3

